### PR TITLE
DATAENG-1778 #comment new batch_migrator code

### PR DESCRIPTION
**Pull request by ArturoAditazz**
I still need to make some cleanup and add tests and do a little more of end-to-end testing but wanted you to take a look at the approach and let me know if you have any feedback.

I'll do more commits later today with tests, debug logging, code comments, and small fixes.

### CODE REVIEW

File: src/main/java/com/aditazz/batch/migrator/config/SecurityAuthorizer.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package com.aditazz.batch.migrator.config;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.access.prepost.PreAuthorize;
+
+@Configuration()
+public class SecurityAuthorizer {
+
+    @PreAuthorize("hasAnyRole('ROOM_FULL', 'ROOM_WRITE') and " +
```
Comment (GermanAditazz) [inline src/main/java/com/aditazz/batch/migrator/config/SecurityAuthorizer.java]: There are no more "FULL" roles, only CREATE, READ, UPDATE, DELETE as far as I remember.

File: src/main/java/com/aditazz/batch/migrator/domain/paginated/RoomPaginated.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package com.aditazz.batch.migrator.domain.paginated;
+
+import com.aditazz.commons.domain.baas.Room;
+
+import java.io.Serializable;
+import java.util.List;
+
+public class RoomPaginated implements Serializable
```
Comment (GermanAditazz) [inline src/main/java/com/aditazz/batch/migrator/domain/paginated/RoomPaginated.java]: Shouldn't this extend `PaginatedList` also?

File: src/main/java/com/aditazz/batch/migrator/processors/ErrorHandlerProcessor.java [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        return result;
+    }
+
+    /**
+     *
+     * @param is
+     * @return
+     * @throws IOException
+     */
+    private String convertStreamToString(InputStream is) throws IOException {
```
Comment (GermanAditazz) [inline src/main/java/com/aditazz/batch/migrator/processors/ErrorHandlerProcessor.java]: Consider using the Scanner trick:

    static String convertStreamToString(java.io.InputStream is) {
        java.util.Scanner s = new java.util.Scanner(is).useDelimiter("\\A");
        return s.hasNext() ? s.next() : "";
    }

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      daas-rules-uri: ${DAAS_RULES_URI:localhost:8093}
+      daas-contents-uri: ${DAAS_CONTENTS_URI:localhost:8092}
+      daas-libraries-uri: ${DAAS_LIBRARIES_URI:localhost:8095}
+      daas-tickets-uri: ${DAAS_TICKETS_URI:localhost:9081}
+      consumer-count: 8
+      max-poll-records: 1
+      #### CHANGE TOPIC TO migration-request !!!!
+      queue-endpoint: kafka:rcp-request?groupId=migrator-group&autoOffsetReset=earliest&consumersCount=${aditazz.batch.migrator.consumer-count}&maxPollRecords=${aditazz.batch.migrator.max-poll-records}
+      source-tenant-roles:
+        - RULE_FULL
```
Comment (GermanAditazz) [inline src/main/resources/application.yml]: There are no more "FULL" roles

File: src/main/resources/bootstrap.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  profiles: local
+  cloud:
+    config:
+      enabled: false
+      discovery:
+        enabled: false
+
+eureka:
+  client:
+    enabled: false
```
Comment (GermanAditazz) [inline src/main/resources/bootstrap.yml]: Where is eureka client configuration enabled=true?

File: src/test/java/com/aditazz/batch/migrator/integration/rest/v1/RestITBaseWithDaasMockSupport.java [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+                    if(obj == null) throw new RuntimeException("Invalid mock object in " +  mock.getName());
+                    return marshalToJson(obj);
+                }catch (RemoteDaasException ex){
+                    return setHttpException(exchange, ex);
+                }
+            }
+        });
+    }
+
+    protected void configureBatchEnqueueOneRcpAgoMock(DaasRemoteFixtureBase fixture){
```
Comment (GermanAditazz) [inline src/test/java/com/aditazz/batch/migrator/integration/rest/v1/RestITBaseWithDaasMockSupport.java]: Is this having wrong names (RcpAgo) or is a copy/paste that can be deleted?

File: README.md [hunk 7]
```diff
@@ -1,2 +1,139 @@
-# baas_migration
-baas_migration
+# BaaS Rooms
```
[the remark this answers is not in the document]
Comment (ArturoAditazz) [reply README.md]: Yeah, I'm still doing some cleanup to get rid of references to rooms (which I used as a base)

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <dependencies>
+        <!-- Aditazz dependencies -->
+        <dependency>
+            <groupId>com.aditazz.daas</groupId>
+            <artifactId>daas-service-security</artifactId>
+            <version>1.0.13-RELEASE</version>
```
[the remark this answers is not in the document]
Comment (ArturoAditazz) [reply pom.xml]: I'll update to newer

File: src/main/java/com/aditazz/batch/migrator/config/SecurityAuthorizer.java (hunk 1 above)
Comment (ArturoAditazz) [reply src/main/java/com/aditazz/batch/migrator/config/SecurityAuthorizer.java]: No problem. I'll remove this class altogether.

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-kafka</artifactId>
+        </dependency>
+
+
+        <!-- Cloud dependencies -->
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-starter-eureka-server</artifactId>
```
[the remark this answers is not in the document]
Comment (ArturoAditazz) [reply pom.xml]: I'll take it off then

File: src/main/java/com/aditazz/batch/migrator/domain/paginated/RoomPaginated.java (hunk 2 above)
Comment (ArturoAditazz) [reply src/main/java/com/aditazz/batch/migrator/domain/paginated/RoomPaginated.java]: This is part of the cleanup, I'll add it to the list.

File: src/main/resources/application.yml (hunk 4 above)
Comment (ArturoAditazz) [reply src/main/resources/application.yml]: That's correct. I'm updating that with CRUD roles that apply to each.

File: src/test/java/com/aditazz/batch/migrator/integration/rest/v1/RestITBaseWithDaasMockSupport.java (hunk 6 above)
Comment (ArturoAditazz) [reply src/test/java/com/aditazz/batch/migrator/integration/rest/v1/RestITBaseWithDaasMockSupport.java]: Part of the cleanup that is in progress

File: src/main/java/com/aditazz/batch/migrator/processors/ErrorHandlerProcessor.java (hunk 3 above)
Comment (ArturoAditazz) [reply src/main/java/com/aditazz/batch/migrator/processors/ErrorHandlerProcessor.java]: I'll remove this and other REST-related methods.

File: src/main/resources/bootstrap.yml (hunk 5 above)
Comment (ArturoAditazz) [reply src/main/resources/bootstrap.yml]: I'll double-check this, this is like this in baas_rooms

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,315 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.aditazz</groupId>
+            <artifactId>data-model-commons</artifactId>
+            <version>2.1.0-RELEASE</version>
+        </dependency>
+        <dependency>
+            <groupId>com.aditazz.daas</groupId>
+            <artifactId>daas-service-commons</artifactId>
+            <version>2.0.0-RELEASE</version>
```
Comment (nemesisAditazz) [inline pom.xml]: upgrade the version

File: src/main/resources/bootstrap.yml (hunk 5 above)
Comment (nemesisAditazz) [reply src/main/resources/bootstrap.yml]: it is only required in application.yml

File: src/test/resources/bootstrap.yml [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+spring:
+  application:
+    name: batch-migrator
+  cloud:
+    config:
+      enabled: false
+      discovery:
+        enabled: false
+
+eureka:
```
Comment (nemesisAditazz) [inline src/test/resources/bootstrap.yml]: it is only required in application.yml

File: src/main/resources/application.yml [hunk 12]
```diff
@@ -0,0 +1,115 @@
+api:
+  version:
+    roadmap: v1,v2
+    active: v1,v2
+    regex: (v\d+.*?)
+
+eureka:
```
Comment (nemesisAditazz) [inline src/main/resources/application.yml]: looks like the dependency for eureka client is missing

**Pull request merged**